GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -989,6 +997,16 @@ - (void)assertValidPresentingViewController {
   }
 }
 
+// Asserts the requested scopes are valid.
+- (void)assertValidScopes:(NSArray<NSString *> *)scopes {
+  if ([scopes containsObject:@"https://www.googleapis.com/auth/verified.age.over18.standard"]) {
```

Comment:
Could we instead make a set of "restricted scopes" and use that in a comparison to see if the provided scopes overlaps with the set of restricted scopes? If there is an overlap, the exception should say what the restricted scopes are and what should be done.

Or maybe you could create a dictionary that maps restricted scope to a class that should be used? For now, it'll just be the one scope, but it could grow overtime (e.g., if we add more verifiable account details that require their own flows). You could use that mapping to tell callers that they have supplied a scope that should use a different class/flow.

Reply:
Decided to follow through with the dictionary suggestion. New method `-[GIDSignIn assertValidScopes:]` will run through the requested scopes and log all scopes (and their classes) that may not be supported by the add scopes flow in GIDSignIn.